src/CommandLineIface.cpp

```diff
@@ -101,8 +160,55 @@ void CommandLineIface::doTranslation() {
     }
 }
 
+void CommandLineIface::downloadRemoteModel(QString modelID) {
+    // fetch model from the internet and wait until it is there
+    connect(&models_, &ModelManager::fetchedRemoteModels, this, [&](){eventLoop_.exit();});
+    models_.fetchRemoteModels();
+    eventLoop_.exec();
+
+    // identify the model we want to download. This is a remote model and we have this complicated lambda function
+    // because i wanted it to be a const & as opposed to a copy of a model
+    const Model& model = [&]() {
+        // Keep track of available models in case we have an error
+        QString errorstr("Unable to find \'" + modelID + "\' in the list of available models. Available models:\n");
+        for (const Model& model : models_.getRemoteModels()) {
+            errorstr = errorstr + model.src + "-" + model.trg + " type: " + model.type + "; To download do -d " + model.shortName + '\n';
+            if (model.shortName == modelID) {
+                return model;
+            }
+        }
+        outputError(errorstr);
+        return models_.getRemoteModels().at(0); // This line will never be executed, since we actually exit in the case of an error
+    }();
+    // Set up progress bar for downloading:
+    QTextStream out(stdout);
+    out << "Downloading " << model.src << "-" << model.trg << " type " << model.type << "...\n";
+    out.flush();
+    connect(&network_, &Network::progressBar, this, [&](qint64 ist, qint64 max) {
+        // taken from https://stackoverflow.com/questions/14539867/how-to-display-a-progress-indicator-in-pure-c-c-cout-printf
+        double percentage = (double)ist/(double)max;
+        int val = (int) (percentage * 100);
+        int lpad = (int) (percentage * PBWIDTH);
+        int rpad = PBWIDTH - lpad;
+        printf("\r%3d%% [%.*s%*s]", val, lpad, PBSTR, rpad, "");
+        fflush(stdout);
+    });
+    // Download the new model. Use eventloop again to prevent premature exit before download is finished
+    connect(&network_, &Network::downloadComplete, this, [&](QFile *file, QString filename) {
+        // We use cout here, as QTextStream out gives a warning about being lamda captured.
+        std::cout << "\nModel downloaded succesffully! You can now invoke it with -m " << modelID.toStdString() << std::endl;
```

Comment:
Double ff 😲

Reply:
I have no idea what you are talking about xD